### ISSUE TITLE
Any way to disable memory? (Remember tabs)

### ISSUE THREAD

**Issue by ilignos** [jenstornell/tabbis.js#9]
Is there any way for the script to not memorize active tabs in other hidden tabs. I only want the visible tabs to have the "active" class.

**Comment by jenstornell** [comment]
I'm working right now with a big update/rewrire. In the new version it will be possible to disable memory completely. Would that solve your problem, or do you mean you want to keep the selected tabs in memory only in the root group?

**Comment by ilignos** [comment]
Hey jen, thanks for your quick reply. I wrote this simple code (based on your demo.html), I have a textbox with class="text_student" and i change its value everytime the active class="student" changes. But because of "tabs memory" i get a value of AlexGeorge or JohnDonald etc instead of Alex or Donald. Is there any quick fix to this issue (i want to keep the same class name="student" without a different id)

Simply put I only want the visible active tabs to have the "active" class.
Here's my full code :)

`<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
    <title>Tabbis</title>
    <style>
        * {
            padding: 0;
            margin: 0;
            font-family: arial;
        }
        body {
            padding: 0 40px;
        }
        h2 {
            margin: 20px 0;
        }
        [data-tabs] {
            display: flex;
            background: #eee;
            padding: 10px;
            border: 2px solid #ccc;
        }
        [data-tabs] > * {
            padding: 5px 20px;
            cursor: pointer;
        }
        [data-tabs] .active {
            background: #31a3e8;
            color: #fff;
            border-radius: 5px;
        }
        [data-panes] > * {
            display: none;
            border: 2px solid #ccc;
            border-top: none;
            padding: 20px;
        }
        [data-panes] > .active {
            display: block;
        }
    </style>
</head>
<body>

<h2>Nested</h2>

<div class="school" data-tabs>
    <div>School 1</div>
    <div>School 2</div>
    <div>School 3</div>
</div>

<div data-panes>
    <div>nothing...</div>
    <div>
        <div class="grade" data-tabs>
            <div>1st grade</div>
            <div>2nd grade</div>
        </div>

        <div data-panes>
            <div>
                <div class="student" data-tabs>
                    <div>John</div>
                    <div>Alex</div>
                </div>

                <div data-panes>
                    <div>Pane6</div>
                    <div>Pane7</div>
                </div>
            </div>
            <div>
                <div class="student" data-tabs>
                    <div>George</div>
                    <div>Donald</div>
                </div>

                <div data-panes>
                    <div>Pane8</div>
                    <div>Pane9</div>
                </div>
            </div>
        </div>
    </div>
    <div>nothing ..</div>
</div>

<input type="text" name="text_school" class="text_school" >
<input type="text" name="text_grade" class="text_grade" >
<input type="text" name="text_student" class="text_student" >

<script>
    $(function(){

        $('.student').click(function () {

            var school_name = $('.school').find('div.active').text();
            var grade_name = $('.grade').find('div.active').text();
            var student_name = $('.student').find('div.active').text();

            $(".text_school").val(school_name);
            $(".text_grade").val(grade_name);
            $(".text_student").val(student_name);
        });
    });

</script>
<script src="tabbis.js"></script>
<script>
    var tabs = tabbis.init({
        tabGroup: "[data-tabs]",
        paneGroup: "[data-panes]",
        tabActive: "active",
        paneActive: "active",
        callback: function(tab, pane) {
            console.log(tab);
            console.log(pane);
        }
    });
</script>
</body>
</html>`

**Comment by ilignos** [comment]
Hey jen, thanks for your quick reply. I wrote this simple code (based on your demo.html).
Simply put I only want the visible active tabs to have the "active" class.
Here's my full code :)

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
    <title>Tabbis</title>
    <style>
        * {
            padding: 0;
            margin: 0;
            font-family: arial;
        }
        body {
            padding: 0 40px;
        }
        h2 {
            margin: 20px 0;
        }
        [data-tabs] {
            display: flex;
            background: #eee;
            padding: 10px;
            border: 2px solid #ccc;
        }
        [data-tabs] > * {
            padding: 5px 20px;
            cursor: pointer;
        }
        [data-tabs] .active {
            background: #31a3e8;
            color: #fff;
            border-radius: 5px;
        }
        [data-panes] > * {
            display: none;
            border: 2px solid #ccc;
            border-top: none;
            padding: 20px;
        }
        [data-panes] > .active {
            display: block;
        }
    </style>
</head>
<body>

<h2>Nested</h2>

<div class="school" data-tabs>
    <div>School 1</div>
    <div>School 2</div>
    <div>School 3</div>
</div>

<div data-panes>
    <div>nothing...</div>
    <div>
        <div class="grade" data-tabs>
            <div>1st grade</div>
            <div>2nd grade</div>
        </div>

        <div data-panes>
            <div>
                <div class="student" data-tabs>
                    <div>John</div>
                    <div>Alex</div>
                </div>

                <div data-panes>
                    <div>Pane6</div>
                    <div>Pane7</div>
                </div>
            </div>
            <div>
                <div class="student" data-tabs>
                    <div>George</div>
                    <div>Donald</div>
                </div>

                <div data-panes>
                    <div>Pane8</div>
                    <div>Pane9</div>
                </div>
            </div>
        </div>
    </div>
    <div>nothing ..</div>
</div>

<input type="text" name="text_school" class="text_school" >
<input type="text" name="text_grade" class="text_grade" >
<input type="text" name="text_student" class="text_student" >

<script>
    $(function(){

        $('.student').click(function () {

            var school_name = $('.school').find('div.active').text();
            var grade_name = $('.grade').find('div.active').text();
            var student_name = $('.student').find('div.active').text();

            $(".text_school").val(school_name);
            $(".text_grade").val(grade_name);
            $(".text_student").val(student_name);
        });
    });

</script>
<script src="tabbis.js"></script>
<script>
    var tabs = tabbis.init({
        tabGroup: "[data-tabs]",
        paneGroup: "[data-panes]",
        tabActive: "active",
        paneActive: "active",
        callback: function(tab, pane) {
            console.log(tab);
            console.log(pane);
        }
    });
</script>
</body>
</html>

```
I have a textbox with class="text_student" and i change its value everytime the active class="student" changes. But because of "tabs memory" i get a value of AlexGeorge or JohnDonald etc instead of Alex or Donald. Is there any quick fix to this issue (i want to keep the same class name="student" without using a different id)

here's the output
![CpWz](https://user-images.githubusercontent.com/57485402/74235921-c2055e00-4cd8-11ea-8f14-2ea8abbcb6e3.png)

**Comment by jenstornell** [comment]
In this case you probably should use the callback. Instead of `console.log(tab)` you can do something like this:

```js
callback: function(tab, pane) {
  const title = tab.innerHTML;
  const key = tab.parentNode.getAttribute('class');

  document.querySelector('.text_' + key).value = title;
}
```

The HTML I used for it is this:

```html
<div data-tabs class="school">
  <div>Tab1</div>
  <div data-active>Tab2</div>
  <div>Tab3</div>
</div>

<div data-panes>
  <div>Pane1</div>
  <div>
    <div data-tabs class="grade">
      <div>Tab4</div>
      <div data-active>Tab5</div>
    </div>

    <div data-panes>
      <div>Pane4</div>
      <div>
        <div data-tabs>
          <div>Tab6</div>
          <div data-active>Tab7</div>
        </div>

        <div data-panes>
          <div>Pane6</div>
          <div>Pane7</div>
        </div>
      </div>
    </div>
  </div>
  <div>Pane3</div>
</div>

<input type="text" class="text_school">
<input type="text" class="text_grade">
```

I hope it helps.

**Comment by ilignos** [comment]
Update: Thank you so much for your answer, it worked :)

**Comment by jenstornell** [comment]
@ilignos Nice! I'll close this issue now then.
